synthesizer/src/process/stack/inclusion/mod.rs

```diff
@@ -95,36 +94,46 @@ impl<N: Network, B: BlockStorage<N>> Query<N, B> {
             Self::VM(block_store) => {
                 block_store.get_program(program_id)?.ok_or_else(|| anyhow!("Program {program_id} not found in storage"))
             }
+            #[cfg(not(feature = "wasm"))]
             Self::REST(url) => match N::ID {
                 3 => Ok(Self::get_request(&format!("{url}/testnet3/program/{program_id}"))?.json()?),
                 _ => bail!("Unsupported network ID in inclusion query"),
             },
+            #[cfg(feature = "wasm")]
+            _ => bail!("External API calls not supported from WASM"),
         }
     }
 
     /// Returns the current state root.
     pub fn current_state_root(&self) -> Result<N::StateRoot> {
         match self {
             Self::VM(block_store) => Ok(block_store.current_state_root()),
+            #[cfg(not(feature = "wasm"))]
             Self::REST(url) => match N::ID {
                 3 => Ok(Self::get_request(&format!("{url}/testnet3/latest/stateRoot"))?.json()?),
                 _ => bail!("Unsupported network ID in inclusion query"),
             },
+            #[cfg(feature = "wasm")]
+            _ => bail!("External API calls not supported from WASM"),
         }
     }
 
     /// Returns a state path for the given `commitment`.
     pub fn get_state_path_for_commitment(&self, commitment: &Field<N>) -> Result<StatePath<N>> {
         match self {
             Self::VM(block_store) => block_store.get_state_path_for_commitment(commitment),
+            #[cfg(not(feature = "wasm"))]
             Self::REST(url) => match N::ID {
                 3 => Ok(Self::get_request(&format!("{url}/testnet3/statePath/{commitment}"))?.json()?),
                 _ => bail!("Unsupported network ID in inclusion query"),
             },
+            #[cfg(feature = "wasm")]
+            _ => bail!("External API calls not supported from WASM"),
         }
     }
 
     /// Performs a GET request to the given URL.
+    #[cfg(not(feature = "wasm"))]
```

Comment:
The reqwest::blocking feature is what prevents wasm from compiling (wasm does not allow thread blocking) and thus the reason for this change. An associated macro was designed for wasm network calls to download the parameters but unfortunately it isn't viable due to the fact that the method called relies in javascript microtasks which make the required calls only after all the code specified in the macro runs, leading to an empty set of parameters (and the way its currently designed, small memory leaks). https://github.com/AleoHQ/snarkVM/blob/testnet3/parameters/src/macros.rs#L175